cmd/client/wallet/main.go

```diff
@@ -925,5 +945,13 @@ func submitTransaction(tx *types.Transaction, walletNode *node.Node, shardID uin
 	// FIXME (leo): how to we know the tx was successful sent to the network
 	// this is a hacky way to wait for sometime
 	time.Sleep(3 * time.Second)
```

Comment:
why wait for 3 seconds? the shown balance should have clear meaning whether it's before and after the transaction is applied. waiting for a few seconds will just make the result ambiguous. I think we just show the balance immediately, so it's always the balance before the transaction.

Reply:
I didn't add the 3 seconds wait logic but from your comment will add commit to remove it.

Reply:
This seems to have been added to wait for transactions to be committed in a block. Immediately calling get balance after submitting txn would get the balance before txn is actually processed.